google/cloud/compute/BUILD.bazel

```diff
@@ -92,94 +93,131 @@ service_dirs = [
     "target_tcp_proxies/v1/",
     "target_vpn_gateways/v1/",
     "url_maps/v1/",
-    "v1/",
     "vpn_gateways/v1/",
     "vpn_tunnels/v1/",
-    "zone_operations/v1/",
     "zones/v1/",
 ]
 
-proto_dirs = service_dirs + [d + "internal/" for d in service_dirs]
+operation_service_dirs = [
+    "global_operations/v1/",
+    "global_organization_operations/v1/",
+    "region_operations/v1/",
+    "zone_operations/v1/",
+]
 
+# Internal protos target
 filegroup(
-    name = "protos",
-    srcs = glob([d + "*.proto" for d in proto_dirs]),
+    name = "internal_protos",
+    srcs = glob(["v1/internal/*.proto"]),
 )
 
 proto_library(
-    name = "google_cloud_cpp_compute_proto",
-    srcs = [":protos"],
+    name = "google_cloud_cpp_compute_internal_proto",
+    srcs = [":internal_protos"],
     visibility = ["//:__pkg__"],
     deps = [
-        "@com_google_googleapis//google/api:annotations_proto",
-        "@com_google_googleapis//google/api:client_proto",
-        "@com_google_googleapis//google/api:field_behavior_proto",
-        "@com_google_googleapis//google/api:http_proto",
-        "@com_google_googleapis//google/api:resource_proto",
-        "@com_google_googleapis//google/api:routing_proto",
         "@com_google_googleapis//google/cloud:extended_operations_proto",
         "@com_google_protobuf//:any_proto",
-        "@com_google_protobuf//:duration_proto",
-        "@com_google_protobuf//:empty_proto",
-        "@com_google_protobuf//:field_mask_proto",
-        "@com_google_protobuf//:struct_proto",
-        "@com_google_protobuf//:timestamp_proto",
     ],
 )
 
-load(
-    "@com_google_googleapis_imports//:imports.bzl",
-    "cc_proto_library",
-)
-
 cc_proto_library(
-    name = "google_cloud_cpp_compute_cc_proto",
-    deps = [":google_cloud_cpp_compute_proto"],
-)
-
-src_dirs = service_dirs + [d + "internal/" for d in service_dirs]
-
-filegroup(
-    name = "srcs",
-    srcs = glob([d + "*.cc" for d in src_dirs]),
+    name = "google_cloud_cpp_compute_internal_cc_proto",
+    deps = [":google_cloud_cpp_compute_internal_proto"],
 )
 
-filegroup(
-    name = "hdrs",
-    srcs = glob([d + "*.h" for d in src_dirs]),
-)
-
-filegroup(
-    name = "mocks",
-    srcs = glob([d + "mocks/*.h" for d in service_dirs]),
-)
+# Service proto targets
+[filegroup(
+    name = service.replace("/v1/", "_proto_srcs"),
+    srcs = glob([service + "*.proto"]),
+) for service in service_dirs + operation_service_dirs]
```

Comment:
Should we just have one loop for "service proto targets" that instantiates all the targets, rather than a loop for each target?